File: Command/CSharp/Editor.cs
namespace CSharp
{
  class Editor
  {
    public string Text { get; set; }

    /// <summary>
    /// Returns the selected text from the start index and for the specified length.
    /// </summary>
    /// <param name="startIndex"></param>
    /// <param name="length"></param>
    /// <returns>String if successful or null otherwise.</returns>
    public string GetSelection( int startIndex, int length )
    {
      if ( string.IsNullOrEmpty( Text ) ||
        length <= 0 ||
        startIndex + length - 1 >= Text.Length )
      {
        return null;
      }
      return Text.Substring( startIndex, length );
    }

    /// <summary>
    /// Deletes the selected text from the start index and for the specified length.
    /// </summary>
    /// <param name="startIndex"></param>
    /// <param name="length"></param>
    /// <returns>The modified string.</returns>
    public string DeleteSelection( int startIndex, int length )
    {
      return Text.Remove( startIndex, length );
    }

    /// <summary>
    /// Deletes the specified text.
    /// </summary>
    /// <param name="text"></param>
    /// <returns>The modified string.</returns>
    public string DeleteSelection( string text )
    {
      return Text.Replace( text, "" ); // or ReplaceSelection( text, "" );
    }

    /// <summary>
    /// Replaces the <c>existingText</c> with <c>newText</c>
    /// </summary>
    /// <param name="existingText"></param>
    /// <param name="newText"></param>
    /// <returns>The modified string.</returns>
    public string ReplaceSelection( string existingText, string newText )
    {
      return Text.Replace(existingText, newText);
    }

    /// <summary>
    /// Inserts the text at the specified index.
    /// </summary>
    /// <param name="startIndex"></param>
    /// <param name="text"></param>
    /// <returns>The modified string.</returns>
    public string InsertSelection( int startIndex, string text )
    {
      return Text.Insert( startIndex, text );
    }
  }
}

File: Command/Cpp/Editor.cpp
#include "Editor.h"

std::string Editor::GetSelection(int startIndex, int length)
{
  if (strText.empty() ||
    length <= 0 ||
    startIndex + length - 1 >= (int)strText.length())
  {
    return nullptr;
  }
  return strText.substr(startIndex, length);
}

std::string Editor::DeleteSelection(int startIndex, int length)
{
  return strText.erase(startIndex, length);
}

std::string Editor::DeleteSelection(std::string text)
{
  std::string str = strText;
  std::size_t findPos = str.find(text);

  while (findPos != std::string::npos)
  {
    str = DeleteSelection(findPos, text.length());
    findPos = str.find(text);
  }

  return str;
}

std::string Editor::ReplaceSelection(std::string existingText, std::string newText)
{
  std::string str = strText;
  std::size_t findPos = str.find(existingText);

  while (findPos != std::string::npos)
  {
    str = DeleteSelection(findPos, existingText.length());
    findPos = str.find(existingText);
  }

  return str;
}

std::string Editor::InsertSelection(int startIndex, std::string text)
{
  return strText.insert(startIndex, text);
}
